### DISCUSSION TITLE
Configuring Renovate to use a separate cache of cloned repositories

### DISCUSSION THREAD

**Discussion by jamietanna** [renovatebot/renovate#32723] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub.com, Mend Renovate 7.5.0

### Please tell us more about your question or problem

While looking at improving the performance of our deployment of Renovate, we've noticed that some of our particularly "heavy" repositories - like https://github.com/elastic/kibana - take a significant chunk of time to clone.

When Renovate has a pre-warmed copy of the repository (i.e if the K8s pod isn't recently redeployed) the `git fetch` can be reasonable, but when using a fresh clone, this can be significantly increased.

We're currently using Mend Renovate Community Edition, but are also looking at whether there are improvements we can make with pure Renovate OSS.

As an organisation, we have some local mirrors of some of our "heavy" repositories, which allow us in CI to perform steps such as:

```sh
git clone --reference /path/to/disk/elastic/kibana.git git@github.com:elastic/kibana.git .
```

We were wondering if there's the capability / appetite to add the capability of, somewhat similar to https://github.com/renovatebot/renovate/discussions/24985, providing the ability to an external location for repositories.

We are not currently using https://docs.renovatebot.com/self-hosted-configuration/#repositorycache, Redis or any other caching configuration

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
What's your experience with "significant chunk of time"?

```
$ time git clone --filter=blob:none https://github.com/elastic/kibana
Cloning into 'kibana'...
remote: Enumerating objects: 1716023, done.
remote: Counting objects: 100% (358/358), done.
remote: Compressing objects: 100% (248/248), done.
remote: Total 1716023 (delta 185), reused 206 (delta 110), pack-reused 1715665 (from 1)
Receiving objects: 100% (1716023/1716023), 391.33 MiB | 38.88 MiB/s, done.
Resolving deltas: 100% (1021332/1021332), done.
remote: Enumerating objects: 81384, done.
remote: Counting objects: 100% (57624/57624), done.
remote: Compressing objects: 100% (44943/44943), done.
remote: Total 81384 (delta 22564), reused 12686 (delta 12681), pack-reused 23760 (from 1)
Receiving objects: 100% (81384/81384), 720.93 MiB | 25.94 MiB/s, done.
Resolving deltas: 100% (28528/28528), done.
Updating files: 100% (83102/83102), done.

real    1m18.774s
```

e.g. is the 78s I saw above on my PC the type of time you're trying to optimize?

**Reply by jamietanna** [reply]
Exactly, yes - we're seeing an average of ~105s in our environment i.e.

```json
{
  "hostname": "...",
  "level": 20,
  "logContext": "...",
  "msg": "Repository timing splits (milliseconds)",
  "name": "renovate",
  "pid": 1696696,
  "repository": "elastic/kibana",
  "splits": {
    "extract": 49799,
    "init": 106766,
    "lookup": 84992,
    "onboarding": 2,
    "update": 636981
  },
  "time": "2024-12-18T09:27:19.808Z",
  "total": 884569,
  "v": 0
}
```

It's possible that adding these `--reference` paths wouldn't do anything, as I think it may be less up-to-date than Renovate's copy of it (given how often Renovate runs vs the cache gets updated)

**Reply by viceice** [reply]
what about the existing `persistRepoData` option?

https://docs.renovatebot.com/self-hosted-configuration/#persistrepodata
